pkg/ccl/backupccl/restore_data_processor.go

```diff
@@ -343,6 +351,12 @@ func (rd *restoreDataProcessor) runRestoreWorkers(
 					return done, nil
 				}
 
+				ctx := logtags.AddTag(ctx, "restore-span", fmt.Sprintf("%s", hex.EncodeToString(entry.Span.Key)))
```

Comment:
nit:

```suggestion
				ctx := logtags.AddTag(ctx, "restore-span", hex.EncodeToString(entry.Span.Key))
```

Reply:
On further consideration, I think this is actually very much not allowed w.r.t. security -- the pprof profile endpoints shouldn't be able to exfiltrate user data. I switched this to the span index (progressidx) in the master version of this change, and will backport that one instead I think.

https://github.com/cockroachdb/cockroach/pull/119572